### PR TITLE
add reply and showing attachment pic

### PR DESCRIPTION
**Pull request by Ketteiteki**
Fixes https://github.com/MangoInstantMessenger/MangoMessengerAPI/issues/387

Implements missing features

### CODE REVIEW

File: MangoAPI.Client/src/app/components/chats/chats.component.html [hunk 1]
```diff
@@ -327,7 +363,47 @@ <h5>Chats</h5>
...
+        <svg
+          class="reply-svg"
+          fill="#adb5bd"
+          height="45"
+          width="45"
+          viewBox="0 0 45 45"
+          xmlns="http://www.w3.org/2000/svg"
+        >
+          <path
+            d="M10.53 5.03a.75.75 0 10-1.06-1.06l-6.25 6.25a.75.75 0 000 1.06l6.25 6.25a.75.75 0 101.06-1.06L5.56 11.5H17a3.248 3.248 0 013.25 3.248v4.502a.75.75 0 001.5 0v-4.502A4.748 4.748 0 0017 10H5.56l4.97-4.97z"
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/chats/chats.component.html:377]: Could we move this svg to a separate file? It doesn't look fine right now

File: MangoAPI.Client/src/app/components/chats/chats.component.html [hunk 2]
```diff
@@ -327,7 +363,47 @@ <h5>Chats</h5>
           </li>
         </ul>
       </div>
-
+      <div *ngIf="_replyStateService.reply" class="reply">
+        <!-- Uploaded to: SVG Repo, www.svgrepo.com, Generator: SVG Repo Mixer Tools -->
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/chats/chats.component.html:367]: Clear this comment please

File: MangoAPI.Client/src/app/components/chats/chats.component.ts [hunk 3]
```diff
@@ -144,8 +174,32 @@ export class ChatsComponent implements OnInit, OnDestroy {
...
+    this._replyStateService.setReply(replyEntity);
+    this.scrollToEnd();
+  }
+
+  async imageSelected(event: any) {
     this.messageAttachment = event.target.files[0];
+
+    var reader = new FileReader();
+  reader.onload = () => {
+    console.log(reader.result);
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/chats/chats.component.ts:194]: Remove this console log

File: MangoAPI.Client/src/app/components/chats/chats.component.ts [hunk 4]
```diff
@@ -275,8 +329,13 @@ export class ChatsComponent implements OnInit, OnDestroy {
...
 
     this.activeChatId = chatId;
-    this.activeChat = this.chats.filter((x) => x.chatId === this.activeChatId)[0];
+    const newActiveChat = this.chats.filter((x) => x.chatId === this.activeChatId)[0];
+    this.activeChat = newActiveChat;
+
+    this.activeChatBehaviorSubject.next(newActiveChat);
+
     this.getChatMessages(this.activeChatId);
+    this.scrollToEnd();
```
Comment (kolosovpetro) [inline MangoAPI.Client/src/app/components/chats/chats.component.ts:338]: Please fix method `this.scrollToEnd()` too, seems it scrolls not to exactly end of last message

**Review by kolosovpetro — CHANGES_REQUESTED** [review]
Few minor changes

File: MangoAPI.Client/src/app/components/chats/chats.component.html (hunk 2 above)
Comment (Ketteiteki) [reply MangoAPI.Client/src/app/components/chats/chats.component.html:367]: it doesn't work without the comment

File: MangoAPI.Client/src/app/components/chats/chats.component.ts (hunk 4 above)
Comment (Ketteiteki) [reply MangoAPI.Client/src/app/components/chats/chats.component.ts:338]: No, it scrolls to exactly end but it do it when pictures are not yet uploaded. When pictures become uploaded scrollbar goes up

File: MangoAPI.Client/src/app/components/chats/chats.component.html (hunk 2 above)
Comment (kolosovpetro) [reply MangoAPI.Client/src/app/components/chats/chats.component.html:367]: > it doesn't work without the comment

Are you kidding me? XD 

So, it is like Linus Torvalds' comments, without them Linux kernel will not work :D

**Review by kolosovpetro — APPROVED** [review]
Looks good to me, thanks

**Pull request merged**